src/generate/apiModel.ts

```diff
@@ -72,9 +72,23 @@ export class ApiModel extends ApiMetadata {
     if (!this.model) {
       throw new Error("Cannot update the name before the model is loaded");
     }
-    this.name = new Name(
-      (this.model.encodes as model.domain.WebApi)?.name.value()
+
+    // class name is defaulted to title
+    let className = (this.model.encodes as model.domain.WebApi)?.name.value();
+
+    // If user defines custom class name under `x-salesforce-sdk-class-name`, use that instead
+    const customClassNameArr = this.model.encodes.customDomainProperties.filter(
+      (customProperty) => customProperty.name.toString() === CUSTOM_NAME_FIELD
     );
+    if (customClassNameArr.length > 0) {
+      // NOTE: there can be multiple instances of `x-salesforce-sdk-class-name` defined, we take the value of the first one
+      const customClassName = customClassNameArr[0].extension
+        .toString()
+        .split("=")[1];
```

Comment:
What is the purpose of split('=') here? How is the '=' symbol used while getting the custom title?

Reply:
There was a better way to get the custom field so I updated this part, but originally the string we were splitting on looked like: `scalar_1:http://www.w3.org/2001/XMLSchema#string=Custom Shop API title`, and we'd split on the equals sign to grab the `Custom Shop API title`